core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -53,47 +48,25 @@ MetadataTableType metadataTableType() {
     return MetadataTableType.ALL_DATA_FILES;
   }
 
-  public static class AllDataFilesTableScan extends BaseFilesTableScan {
+  public static class AllDataFilesTableScan extends BaseAllFilesTableScan {
```

Comment:
I added `BaseAllFilesTableScan` for all aggregated file tables as I think it is slightly cleaner. That way, we can properly override the planning method in the parent class and consume in all children.